docs/apis/calcite.md

```diff
@@ -0,0 +1,529 @@
+Apache Apex,unified stream processing engine enables application developers to process data at very high throughput with low latency. Though type of data demands different type of
+processing requirements but SQL remains way to process input data .To ensure that existing ETL developers,developers who are well versed with Database applications adopt stream processing appliction development with ease, integration of SQL  with Apex was needed. Being popular and Apache project, Calcite was chosen for an integration and to facilitate use of SQL with Apex.
+
+## APEX-Calcite Integrtion
+ Apache Calcite is highly customizable engine for parsing and planning queries on relational data from various data sources . Apache Calcite provides functionality of storage independent query optimization and ways to integrate them in other frameworks which would like to take advantage and expose SQL capability to their users.
+
+Particulary in Apex-Calcite integration, Calcite processes query and then creates Relational Algebra to create processing pipelines. These relational algebra processing pipelines
+are coverted DAG with set of operators to perform business logic on stream data.
+
+![image alt text](images/calcite-apex.png)
+
+
+Above figure explains how SQL query gets converted to Apex DAG.
+
+1. The received query from a user is parsed by Calcite’s query parser.
+2. The query parser returns the SQL Node Tree to Query Planner and optimizer which converts the SQL Node Tree to Relation Expression Tree.
+3. This Relation Expression Tree is received by Apache Apex’s SQL module to finally convert to an Apex DAG having series of operators.
+
+One peculiarity that can be seen in above example is the data source, and data destination are not RDBMS systems. This is because Calcite allows Apex to register table sources and destinations as anything which can return a row type results. So a “scan” relational expression gets converted to “KafkaInputOperator + ParseOperator”, a result of which is series of POJO reflecting a Row Type. Similarly, the “insert” Relational Expression translated to “FormatOperator + FileOutputOperator”.
+
+
+For details about Apache Calcite, please read at [Apache Calcite Website](https://calcite.apache.org/) 
+
+## SQL Application with Model file as input:
```

Comment:
Please have section which says following:
Title: "SQL APIs for Apache Apex"
Content:
"Listed below are couple of java APIs which can be used by SQL/Apex users to create a DAG using just a SQL statement.
1. Simple SQL Application with SQL API:
2. Pure Style SQL Application:
3. Fusion Style SQL Application: 
"